Missing include

The `functional` header is missing from the cpp file.

Build log:
```
g++ -flto -O3 -o cfg-checker main.cpp \
	-std=c++11 -Wall -Wextra -Wpedantic -Werror -Wno-unused-parameter
main.cpp: In function ‘int main(int, char**)’:
main.cpp:444:10: error: ‘function’ is not a member of ‘std’
     std::function<size_t(const std::shared_ptr<sentential_form> key)>,
          ^~~~~~~~
main.cpp:444:10: note: suggested alternative: ‘is_function’
     std::function<size_t(const std::shared_ptr<sentential_form> key)>,
          ^~~~~~~~
          is_function
main.cpp:444:10: error: ‘function’ is not a member of ‘std’
main.cpp:444:10: note: suggested alternative: ‘is_function’
     std::function<size_t(const std::shared_ptr<sentential_form> key)>,
          ^~~~~~~~
          is_function
main.cpp:444:69: error: template argument 2 is invalid
     std::function<size_t(const std::shared_ptr<sentential_form> key)>,
                                                                     ^
main.cpp:444:70: error: expected unqualified-id before ‘,’ token
     std::function<size_t(const std::shared_ptr<sentential_form> key)>,
                                                                      ^
main.cpp:445:18: error: qualified-id in declaration before ‘<’ token
     std::function<bool(
                  ^
main.cpp:457:6: error: expected primary-expression before ‘,’ token
     },
      ^
main.cpp:472:3: error: ‘visited’ was not declared in this scope
   visited.insert(s);
   ^~~~~~~
Makefile:6: recipe for target 'cfg-checker' failed
make: *** [cfg-checker] Error 1
```

--- a/main.cpp
+++ b/main.cpp
@@ -1,5 +1,6 @@
 #include <cstdint>
 #include <fstream>
+#include <functional>
 #include <iostream>
 #include <memory>
 #include <queue>

